### ISSUE TITLE
ignoring auth URL when host setted without dots

### ISSUE THREAD

**Issue by Dryymoon** [jcmoraisjr/haproxy-ingress#817]
**Description of the problem**

We used custom auth proxy microservice.
Before and in release 0.13.0-snapshot.2 all work fine.
But after release 0.13.0-snapshot.3 released - Haproxy ignoring auth url and auth proxy dont work.

Partial config of target restricted service
```
...
ingress:
  enabled: true
  annotations:
    kubernetes.io/ingress.class: haproxy
    ingress.kubernetes.io/auth-url: "http://ms-authentication/v1/haproxy-firewall"
...
```
Error:
```
backend.go:159] ignoring auth URL with an invalid domain on ingress 'default/some-restricted-service': lookup ms-authentication/v1: no such host
```

```ms-authentication``` - real name of custom auth service, service deployed inside kubernetes

**Expected behavior**

Auth proxy works without errors,
All requests to target service should be inspected by auth proxy

**Current behavior**
Haproxy Ingress controller hit error in console and all request to target service go without any authentification by auth service

**Environment information**

HAProxy Ingress version: 
0.13.0-snapshot.2 - works well
0.13.0-snapshot.3, 0.13.0-beta.1 - not work, got error

Ingress objects:

```
ingress:
  enabled: true
  annotations:
    kubernetes.io/ingress.class: haproxy
    ingress.kubernetes.io/auth-url: "http://ms-authentication/v1/haproxy-firewall"
```

**Comment by jcmoraisjr** [comment]
Hi thanks for this reproducer! This was a corner case where a http proto was used, you entered `<domain>/<path>` but `<namespace>/<name>` pattern was extracted. You can work around this in the current version adding a domain suffix, eg `ms-authentication.<your-namespace>`. We'll fix this in the next tag.

**Comment by jcmoraisjr** [comment]
Fixed in `v0.13.0-beta.2`. Also, requests will always be denied if an attempt to configure external authentication fail.
